Sources/RepoToolsCore/RepoActions.swift

```diff
@@ -398,25 +401,41 @@ public enum RepoActions {
         // Extract the downloaded archive
         let extractDir = shell.tmpdir()
         func extract() -> CommandOutput {
-            let lowercasedFileName = fileName.lowercased()
-            if lowercasedFileName.hasSuffix("zip") {
+            var extractionHandler = fetchOptions.extractionHandler.isEmpty ? "default" : fetchOptions.extractionHandler;
+            if extractionHandler == "default" {
+                let lowercasedFileName = fileName.lowercased()
+                if lowercasedFileName.hasSuffix("zip") {
+                    extractionHandler = "zip"
+                } else if (
+                    lowercasedFileName.hasSuffix("tar.gz") ||
+                    lowercasedFileName.hasSuffix("tar") ||
+                    lowercasedFileName.hasSuffix("tgz")) {
+                    extractionHandler = "tar"
+                } else {
+                    fatalError("Cannot extract files other than .zip, .tar, .tar.gz, or .tgz")
+                }
+            }
+
+            if extractionHandler == "zip" {
                 return shell.command(CommandBinary.sh, arguments: [
                     "-c",
                     unzipTransaction(
                         rootDir: extractDir,
                         fileName: escape(download)
                     ),
                 ])
-            } else if lowercasedFileName.hasSuffix("tar.gz") {
+            } else if extractionHandler == "tar" {
                 return shell.command(CommandBinary.sh, arguments: [
                     "-c",
                     untarTransaction(
                         rootDir: extractDir,
                         fileName: escape(download)
                     ),
                 ])
+            } else {
+                let handler = extractionHandler ?? "default"
```

Comment:
Consider moving `ExtractionHandler` to an enum ( `.zip`, `.tar`, etc ), and switching on the kind so it's validated at compile time.